include/bluetooth/conn.h

```diff
@@ -711,8 +711,61 @@ struct bt_conn_oob_info {
 	};
 };
 
+#if defined(CONFIG_BT_SMP_PAIRING_ALLOW)
+/** Info structure for generic_pairing_query */
+struct bt_conn_pairing_auth_rec_info {
+	u8_t bonding_flags:2,
+	     mitm:1,
+	     sc:1,
+	     keypress:1,
+	     ct2:1,
+	     rfu:2;
+};
+
+struct bt_conn_pairing_info {
+	u8_t io_capability;
+	u8_t oob_data_flag;
+	struct bt_conn_pairing_auth_rec_info auth_req;
+	u8_t max_enc_key_size;
+	u8_t init_key_dist;
+	u8_t resp_key_dist;
+};
+#endif /* CONFIG_BT_SMP_PAIRING_ALLOW */
+
 /** Authenticated pairing callback structure */
 struct bt_conn_auth_cb {
+#if defined(CONFIG_BT_SMP_PAIRING_ALLOW)
+	/** @brief Query to proceed incoming pairing or not.
+	 *
+	 *  On any incoming pairing req/rsp this callback will be called for
+	 *  the application to decide whether to allow for the pairing to
+	 *  continue.
+	 *
+	 *  The pairing info received from the peer is passed to assist
+	 *  making the decision.
+	 *
+	 *  As this callback is synchronous the application should return
+	 *  a response value immediately. Otherwise it may affect the
+	 *  timing during pairing. Hence, this information should not be
+	 *  conveyed to the user to take action.
+	 *
+	 *  The remaining callbacks are not affected by this, but do notice
+	 *  that other callbacks can be called during the pairing. Eg. if
+	 *  pairing_confirm is registered both will be called for Just-Works
+	 *  pairings.
+	 *
+	 *  This callback may be unregistered in which case pairing continues
+	 *  as if the Kconfig flag was not set.
+	 *
+	 * Currently BT SSP is not supported.
```

Comment:
Yes, hence the "Currently BT SSP is not supported." line. Should it be more explicit?

Reply:
I wanted to change to that line instead. SSP has not been used in the header so would make sense to define what it means. And otherwise BR/EDR is more clear, as SSP is only relevant for BR/EDR. BT is never used as an abbreviation, we always spell use either Bluetooth LE or BR/EDR.